publishers/community/generic.py

```diff
@@ -163,3 +164,46 @@ def _recursive_enumerate_fields(structure, output_reference, path=''):
     _recursive_enumerate_fields(publication, output)
 
     return OrderedDict(sorted(output.items()))
+
+
+@Register
+def bubble_fields(alert, publication):
+    """This publisher bubbles all requested fields to the top level and deletes everything else.
```

Comment:
Does the publisher really delete non-requested fields? I think it just bubbles all requested fields to the top level.

Reply:
It deletes it from the `publication`, not from the `alert.record`. The way **publishers** work, they are supposed to chain the `publication` from one publisher to the next, adding/transforming fields as they go. This one is "special" because it ignores the previous publishers' work and just throws all of the existing fields away.